### DISCUSSION TITLE
Rendering a Shape on specific feature

### DISCUSSION THREAD

**Discussion by hishamco** [OrchardCMS/OrchardCore#12014] [Ideas]
It's very handy to render shapes when the certain feature is enabled. Can we extend display management to for such scenarios. I'm working on  a feature to display a list in TheAdmin theme but this list should be displayed when specific feature is on, so instead of adding a check to accomplish my task it's good to have an extension methods to let you display the shape when some features are on

**Comment by ns8482e** [comment]
You can use `[Feature]` attribute on any C# class

**Reply by hishamco** [reply]
Yep, but in case I call the shape via TagHelper or through  `DisplayAsync()` it throws an exception if the feature it disabled

**Reply by ns8482e** [reply]
what kind of exception?

**Reply by hishamco** [reply]
`ShapeNotFound`, please refer to my PR #12017 as use case

**Reply by ns8482e** [reply]
Isn't that valid exception? that really needs an attention

**Reply by ns8482e** [reply]
IMHO it's design issue, you really need to have the caller that calls the rendering of the shape also needs to be part of the same feature?  why would you try to render a shape outside of it's feature?

**Reply by hishamco** [reply]
Believe it or not there's a need to find some simple way to allow render a shape if specific feature is on, and hide it automatically when it's off. I don't think adding a setting or make a check everywhere is a proper solution

**Reply by ns8482e** [reply]
@hishamco for #12017 you can place shape in `NavbarTop` from the feature that defines the shape

**Reply by hishamco** [reply]
Could you please elaborate? How the shape will not throw if the feature is disabled?

**Reply by ns8482e** [reply]
It throws exception because you explicitly trying render with New.CulturePicker and shape that you are rendering is not defined in module

**Reply by ns8482e** [reply]
You should use similar  technique, like left menu is injected in layout

you can inject shape into `NavbarTop` from the module that actually defines the shape

**Reply by hishamco** [reply]
> It throws exception because you explicitly trying render with New.CulturePicker and shape that you are rendering is not defined in module

Exactly, that's why I need a way to render nothing if the feature is disabled

> You should use similar technique, like left menu is injected in layout

I think the side menu rendered all the time
